### PR TITLE
Refactor InferenceBackend to Model. Related code cleanup 

### PR DESCRIPTION
**Pull request by GuanLuo**
The commit Rename InferenceBackend to Model (569f7f4ba3441484ada7850643e3d2dc0bcfdb6e) is just name changes and can be skipped

### CODE REVIEW

**Comment by tanmayv25** [comment]
Looks good clean up. We can also move the files from `src/backends/` to `src/core` and remove the backends directory from this repo.

File: src/backends/backend/triton_model.cc [hunk 1]
```diff
@@ -229,11 +216,61 @@ TritonModel::UpdateModelConfig(
...
+  std::unordered_map<std::string, bool> enforce_equal_shape_tensors;
+  for (const auto input : config_.input()) {
+    if (input.is_shape_tensor()) {
+      enforce_equal_shape_tensors.insert({input.name(), true});
+    } else if (!input.allow_ragged_batch() && (GetElementCount(input) == -1)) {
+      enforce_equal_shape_tensors.insert({input.name(), false});
+    }
+  }
+
+  // If 'sequence_batching' is configured use the SequenceBatchScheduler,
```
Comment (CoderHam) [inline src/backends/backend/triton_model.cc:243]: configured, then use

File: src/core/model_config.h [hunk 2]
```diff
@@ -51,15 +51,6 @@ using BackendCmdlineConfig = std::vector<std::pair<std::string, std::string>>;
 using BackendCmdlineConfigMap =
```
Comment (CoderHam) [inline src/core/model_config.h:51]: update copyright

File: src/backends/ensemble/ensemble_model.cc [hunk 3]
```diff
@@ -24,7 +24,7 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
```
Comment (CoderHam) [inline src/backends/ensemble/ensemble_model.cc:24]: Fix copyright

File: src/backends/ensemble/ensemble_model.h [hunk 4]
```diff
@@ -26,38 +26,35 @@
 #pragma once
```
Comment (CoderHam) [inline src/backends/ensemble/ensemble_model.h:26]: Fix copyright

**Review by CoderHam — CHANGES_REQUESTED** [review]
Minor changes else LGTM

**Review by deadeyegoodwin — DISMISSED** [review]

**Review by CoderHam — APPROVED** [review]

**Pull request merged**